client/services/code_preprocessors/PythonCodePreprocessorService.js

```diff
@@ -95,6 +97,52 @@ tie.factory('PythonCodePreprocessorService', [
       return firstLine + '\n' + subsequentLines.join('\n');
     };
 
+    // Adds support for helper functions in student code by
+    // dynamically inserting the StudentCode class name into their code
+    // during the preprocessing phase, allowing it to run normally without
+    // any indication on the student side that this is happening.
+    var _addClassWrappingToStudentHelperFunctions = function(code){
+      while (true){
+        // Returns ['matching string', 'capture group']
+        var regexResult = PYTHON_FUNCTION_DEF_REGEX.exec(code);
+        if (!regexResult) {
+          return code;
+        }
```

Comment:
Mild suggestion: at this point, consider declaring variables matchingString and functionName (and using those) rather than continuing to work with regexResult[0] and regexResult[1]. Easier to understand.